### ISSUE TITLE
I miss you CleanLinks

### ISSUE THREAD

**Issue by jawz101** [diegocr/CleanLinks#172]
sad face

**Comment by diegocr** [comment]
I miss it too ;)

Let's have a look at it when WebExtensions are formally established on stable releases of Firefox.

**Comment by jawz101** [comment]
Have you kept up with the nightlies?

I decided to go ahead and try it when they brought in layout.css.servo.enabled option (blazing fast.)  And then 2 days ago they set extensions.legacy.enabled defaulting to false - which disabled even hybrid extensions.  Suddenly AMO has maybe a hundred or so add-ons that are pure web extensions will install.  NoScript, Lastpass, and Decentraleyes all moved into a disabled bucket on my browser.

And then yesterday my nightly icon was [this](https://s18.postimg.org/sgzvqys1l/firefoxhobbes.png).  Today it was [this](https://i.redd.it/x3jiq3lyrhfz.png).  I think they are hinting we better board the hype train.

Giorgio was [also interviewed](https://blog.mozilla.org/addons/2017/08/01/noscripts-migration-to-webextensions-apis/) on his NoScript work.  I get the feeling they feel their current API's are solidified for v57 and now they're hoping people will start prepping for November.

My worry is for add-ons that might not go through the motions by then might have longer delays if they rely on extension API's that haven't been accounted for yet.  Seems like you have to put in Bugzilla posts to get someone to consider reintroducing API's to allow for particular things.  Like, some people had add-ons which tweaked about:config prefs so now there's a bugzilla issue out there to see about exposing a subset of prefs to be able to be manipulated by an add-on.

Basically, I don't know if you'll lose functionality to do what you did and, if so, would you ignore that piece or submit a request to have a solution re-added?  Because that might mean an even longer wait or less priority by Mozilla devs than someone who prepped before then.  I dunno.

**Comment by diegocr** [comment]
Actually, I have been using Nightly builds since ages ago, until a week or two ago that after an upgrade it refuses to start any longer, literally opening dozens of dialogs telling me that my profile is corrupt or such... even using a fresh profile gives the same behavior, which did remind me to some Windows 95 virus that was doing the same...i have to kill it from the TaskMan 🙄 

Not sure what to think about the icons changing like that... well, the last one as you may know is the former Mozilla Firebird one (Firefox < 1.0)

As for your worries, the main CleanLinks algorithm can be (re-)implemented with vanilla JS and support for DOMContentLoaded or click listeners on the html document, no trouble doing that within a WebExtension i think, we may lose some functionality from advanced features such as _http-observer_, but at least a light version should be fast to implement...or so i hope.

**Comment by jawz101** [comment]
I never quite understood what the http observer was ... even though I activated it anyways.

I think I'm going back to stable, regardless.  I'd rather the few add-ons that were hybrid still work for now.  It's nice to see what nightly has going on but it feels like a war zone of half-baked ideas that may or may not move to stable.  Kinda frustrating.

**Comment by alekksander** [comment]
please re-open. this thread contains important information from the author.

edit:
thank You

**Comment by alekksander** [comment]
@diegocr any thoughts after v.57 has landed?

**Comment by YourFin** [comment]
I'd be happy to help port if it's needed

**Comment by ImSpecial** [comment]
Been using [Skip Redirect](https://addons.mozilla.org/en-US/firefox/addon/skip-redirect/) as a replacement for Clean Links for awhile now and couldn't be happier, I suggest you guys/gals give it a try.

https://github.com/sblask/webextension-skip-redirect

**Comment by jawz101** [comment]
@diegocr [push it.](https://www.youtube.com/watch?v=vCadcBR95oU)

**Comment by diegocr** [comment]
@jawz101 [yup..i know](https://www.youtube.com/watch?v=n4RjJKxsamQ) ;)

**Comment by jawz101** [comment]
@diegocr I like your taste lol
I can never understand what he's saying in the 1st sentence.  I guess it's about places in Russia.  Moskva = Moscow  and some place called Gorky Park.

**Comment by jawz101** [comment]
any update?

**Comment by ioc2e3** [comment]
It’s 2025 rear ...

Still no update ...

.
